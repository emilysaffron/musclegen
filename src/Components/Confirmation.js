import styled from "@emotion/styled";
import ToggleModal from "../Helpers/ToggleModal";
import AddExercise from "../Helpers/AddExercise";
const StyledButton = styled.button`
  border-radius: 20px;
  width: 150px;
  height: 90px;
  padding: 20px;
  margin: 10px;
  font-family: Palatino, URW Palladio L, serif;
  color: black;
  &:hover {
    cursor: pointer;
  }
  outline: none;
`;

const ButtonsContainer = styled.div`
  display: flex;
`;

const Confirmation = ({
  choice,
  repNumber,
  exercise,
  modal,
  toggleModal,
  AddToWorkoutPlan,
  workoutPlan,
}) => {
  const handleCancelClick = () => {
    ToggleModal(toggleModal, modal);
  };
  const handleConfirmClick = () => {
    AddExercise(exercise, repNumber, AddToWorkoutPlan, workoutPlan);
    ToggleModal(toggleModal, modal);
  };

  return (
    <ButtonsContainer>
      <StyledButton onClick={() => handleConfirmClick()}>
        add to plan
      </StyledButton>
      <StyledButton onClick={() => handleCancelClick()}>cancel</StyledButton>
    </ButtonsContainer>
  );
};

export default Confirmation;
