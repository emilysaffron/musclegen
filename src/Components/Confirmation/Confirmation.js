import styled from "@emotion/styled";

const StyledButton = styled.button`
  border-radius: 20px;
  width: 150px;
  height: 90px;
  padding: 20px;
  margin: 10px;
  cursor: pointer;
  outline: none;
`;

const ButtonsContainer = styled.div`
  display: flex;
`;

const Confirmation = ({
  repNumber,
  exercise,
  modal,
  toggleModal,
  AddToWorkoutPlan,
  workoutPlan,
}) => {
  const handleCancelClick = () => {
    toggleModal(!modal);
  };
  const handleConfirmClick = () => {
    let currentWorkoutPlan = workoutPlan.slice();
    let index = currentWorkoutPlan.indexOf(
      exercise + " for " + repNumber + " rep(s)"
    );

    if (index === -1) {
      currentWorkoutPlan.push(exercise + " for " + repNumber + " rep(s)");
    }
    AddToWorkoutPlan(currentWorkoutPlan);

    toggleModal(!modal);
  };

  return (
    <ButtonsContainer>
      <StyledButton onClick={() => handleConfirmClick()}>
        add to plan
      </StyledButton>
      <StyledButton onClick={() => handleCancelClick()}>cancel</StyledButton>
    </ButtonsContainer>
  );
};

export default Confirmation;
