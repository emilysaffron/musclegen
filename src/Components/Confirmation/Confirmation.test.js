import renderer from "react-test-renderer";
import Confirmation from "./Confirmation";

it("Confirmation Snapshot", () => {
  const tree = renderer
    .create(
      <Confirmation
        repNumber={0}
        exercise={""}
        modal={true}
        toggleModal="toggleModal"
        AddToWorkoutPlan="AddToWorkoutPlan"
        workoutPlan=""
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
