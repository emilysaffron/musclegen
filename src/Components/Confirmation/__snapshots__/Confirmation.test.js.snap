// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Snapshot 1`] = `
<div
  className="css-ov1ktg"
>
  <button
    className="css-ljtnwa"
    onClick={[Function]}
  >
    add to plan
  </button>
  <button
    className="css-ljtnwa"
    onClick={[Function]}
  >
    cancel
  </button>
</div>
`;
